### PR TITLE
Chore: Added synchronization mode, database creation from scratch, progress bars, mainnet support

### PR DESCRIPTION
**Pull request by MarMarMot**

### CODE REVIEW

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -1,24 +1,108 @@
...
+    initdb_command="/usr/lib/postgresql/$PG_VERSION/bin/initdb --pgdata=/var/lib/postgresql/$PG_VERSION/main --auth=md5 --auth-local=md5 --auth-host=md5 --username=postgres --pwfile=/tmp/password"
+    sudo -H -u postgres bash -c "$initdb_command"
+}
+
+echo "entrypoint - network: $NETWORK"
+NETWORK_STR="--testnet-magic $PROTOCOL_MAGIC"
+HARDFORK_EPOCH=1
+if [[ "$NETWORK" == "mainnet" ]]; then
+    NETWORK_STR="--mainnet"
+    HARDFORK_EPOCH=208
```
Comment (MarMarMot) [inline docker/entrypoint.sh:71]: @Kammerlo I found these values on google, but couldn't find a description of what they are. Do you know where I can read about it and how I can check that it is correct.

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -1,24 +1,108 @@
...
+
+               show_progress "Node verification: Chunk" $current $total $percent
+          fi
+          sleep 1
+    done
+    echo -e "Node verification: DONE$(printf "%100s")"
+}
+
+node_synchronization() {
+    echo -e "Node synchronization..."
```
Comment (MarMarMot) [inline docker/entrypoint.sh:35]: Check this part carefully, please.

File: docker/entrypoint.sh [hunk 3]
```diff
@@ -1,24 +1,108 @@
...
+    epoch_length=$(jq -r .epochLength $GENESIS_SHELLEY_PATH)
```
Comment (MarMarMot) [inline docker/entrypoint.sh:36]: This code was taken from here:
https://forum.cardano.org/t/check-sync-status-of-node
Please check it carefully

**Review by Kammerlo — COMMENTED** [review]
This looks good, I will test it tomorrow. Since I don't have any time left today. But Good work, Thank you!

**Review by Kammerlo — APPROVED** [review]

**Review by Kammerlo — APPROVED** [review]

**Pull request merged**